lib/astarte/core/generators/triggers/simple_events/simple_event.ex

```diff
@@ -0,0 +1,134 @@
+    as: InterfaceRemovedEventGenerator
+
+  alias Astarte.Core.Generators.Triggers.SimpleEvents.PathCreatedEvent,
+    as: PathCreatedEventGenerator
+
+  alias Astarte.Core.Generators.Triggers.SimpleEvents.PathRemovedEvent,
+    as: PathRemovedEventGenerator
+
+  alias Astarte.Core.Generators.Triggers.SimpleEvents.ValueChangeAppliedEvent,
+    as: ValueChangeAppliedEventGenerator
+
+  alias Astarte.Core.Generators.Triggers.SimpleEvents.ValueChangeEvent,
+    as: ValueChangeEventGenerator
+
+  alias Astarte.Core.Generators.Triggers.SimpleEvents.ValueStoredEvent,
+    as: ValueStoredEventGenerator
+
+  @spec simple_event() :: StreamData.t(SimpleEvent.t())
+  @spec simple_event(keyword :: keyword()) :: StreamData.t(SimpleEvent.t())
+  def simple_event(params \\ []) do
+    params gen all event <- event(),
+                   simple_trigger_id <- simple_trigger_id(),
+                   parent_trigger_id <- parent_trigger_id(),
+                   realm <- realm(),
+                   device_id <- device_id(),
+                   timestamp <- timestamp(),
+                   params: params do
+      %SimpleEvent{
+        event: event,
+        simple_trigger_id: simple_trigger_id,
+        parent_trigger_id: parent_trigger_id,
+        realm: realm,
+        device_id: device_id,
+        timestamp: timestamp
+      }
+    end
+  end
+
+  defp event do
+    [
+      {:device_connected_event, DeviceConnectedEventGenerator.device_connected_event()},
+      {:device_disconnected_event, DeviceDisconnectedEventGenerator.device_disconnected_event()},
+      {:incoming_data_event, IncomingDataEventGenerator.incoming_data_event()},
+      {:value_change_event, ValueChangeEventGenerator.value_change_event()},
+      {:value_change_applied_event,
+       ValueChangeAppliedEventGenerator.value_change_applied_event()},
+      {:path_created_event, PathCreatedEventGenerator.path_created_event()},
+      {:path_removed_event, PathRemovedEventGenerator.path_removed_event()},
+      {:value_stored_event, ValueStoredEventGenerator.value_stored_event()},
+      {:incoming_introspection_event,
+       IncomingIntrospectionEventGenerator.incoming_introspection_event()},
+      {:interface_added_event, InterfaceAddedEventGenerator.interface_added_event()},
+      {:interface_removed_event, InterfaceRemovedEventGenerator.interface_removed_event()},
+      {:interface_minor_updated_event,
+       InterfaceMinorUpdatedEventGenerator.interface_minor_updated_event()},
+      {:device_error_event, DeviceErrorEventGenerator.device_error_event()}
+    ]
+    # NOTE
+    # Using `one_of()` would be the right solution, but StreamData's `specs` do not recognise
+    # an `atom` within a `tuple` as `StreamData` and therefore as a generator.
+    # one_of()
+    |> member_of()
+    |> bind(fn {atom, gen} -> gen |> map(&{atom, &1}) end)
+  end
+
+  defp simple_trigger_id, do: binary()
+  defp parent_trigger_id, do: binary()
```

Comment:
technically, this would be more correct

```suggestion
  defp simple_trigger_id, do: repeatedly(fn -> UUID.uuid4(:raw) end)
  defp parent_trigger_id, do: repeatedly(fn -> UUID.uuid4(:raw) end)
```